diff -up perl-5.36.0/Makefile.SH.0207~ perl-5.36.0/Makefile.SH
--- perl-5.36.0/Makefile.SH.0207~	2022-07-09 01:47:44.037822633 +0200
+++ perl-5.36.0/Makefile.SH	2022-07-09 01:48:42.947273140 +0200
@@ -378,6 +378,7 @@ RUN_PERL = \$(LDLIBPTH) \$(RUN) $perl\$(
 # Macros to invoke a copy of our fully operational perl during the build.
 PERL_EXE = perl\$(EXE_EXT)
 RUN_PERL = \$(LDLIBPTH) \$(RUN) ./perl\$(EXE_EXT) -Ilib -I.
+RUN_PERL_PGO = \$(LDLIBPTH) \$(RUN) ../$perl\$(EXE_EXT)
 !GROK!THIS!
 	;;
 esac
@@ -1671,6 +1672,11 @@ test_reonly test-reonly: test_prep_reonl
 test_porting test-porting: test_prep
 	TEST_ARGS='porting/*.t lib/diagnostics.t' TESTFILE=harness $(RUN_TESTS) choose
 
+# Test perlbench for pgo profiling
+test_pgo: $(PERL_EXE)
+	cd perlbench/ && $(RUN_PERL_PGO) perlbench-runtests
+
+
 !NO!SUBS!
 
 $spitshell>>$Makefile <<!GROK!THIS!
