diff -p -up perl-5.10.0/Configure.pix perl-5.10.0/Configure
--- perl-5.10.0/Configure.pix	2007-12-18 11:47:07.000000000 +0100
+++ perl-5.10.0/Configure	2007-12-21 09:59:10.000000000 +0100
@@ -1263,6 +1263,13 @@ elif test -d c:/. -o -n "$is_os2" ; then
     _exe=".exe"
 fi
 
+: determine libdir name
+: Should be */lib64 on x86-64, sparc64, ppc64, aarch64
+case `arch` in
+  x86_64|ppc64|aarch64) lib='lib64';;
+  *) lib='lib';;
+esac
+
 groupstype=''
 i_whoami=''
 : Trailing extension.  Override this in a hint file, if needed.
@@ -1273,12 +1280,12 @@ libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
 xlibpth='/usr/lib/386 /lib/386'
 : Possible local library directories to search.
-loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
-loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
+loclibpth="/usr/local/$lib /opt/local/$lib /usr/gnu/$lib"
+loclibpth="$loclibpth /opt/gnu/$lib /usr/GNU/$lib /opt/GNU/$lib"
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
+glibpth="/$lib /usr/$lib $xlibpth"
+glibpth="$glibpth /usr/ccs/$lib /usr/ucblib /usr/local/$lib"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
 test -d /usr/lib64         && glibpth="$glibpth /lib64 /usr/lib64 /usr/local/lib64"
@@ -10970,9 +10977,9 @@ else
 		else
 			for net in net socket
 			do
-				if test -f /usr/lib/lib$net$_a; then
-					( ($nm $nm_opt /usr/lib/lib$net$_a | eval $nm_extract) ||  \
-					$ar t /usr/lib/lib$net$_a) 2>/dev/null >> libc.list
+				if test -f /usr/$lib/lib$net$_a; then
+					( ($nm $nm_opt /usr/$lib/lib$net$_a | eval $nm_extract) ||  \
+					$ar t /usr/$lib/lib$net$_a) 2>/dev/null >> libc.list
 					if $contains socket libc.list >/dev/null 2>&1; then
 						d_socket="$define"
 						socketlib="-l$net"
@@ -22223,6 +22230,7 @@ ldflags='$ldflags'
 ldflags_uselargefiles='$ldflags_uselargefiles'
 ldlibpthname='$ldlibpthname'
 less='$less'
+lib='$lib'
 lib_ext='$lib_ext'
 libc='$libc'
 libperl='$libperl'
