diff -up perl-5.30.0/t/io/fs.t.0007~ perl-5.30.0/t/io/fs.t
--- perl-5.30.0/t/io/fs.t.0007~	2019-06-28 12:47:34.915694203 +0200
+++ perl-5.30.0/t/io/fs.t	2019-06-28 14:46:29.055348081 +0200
@@ -257,7 +257,7 @@ isnt($atime, $ut,          'atime: utime
 isnt($mtime, $ut + $delta, 'mtime: utime called with two undefs');
 
 SKIP: {
-    skip "no futimes", 6 unless ($Config{d_futimes} || "") eq "define";
+    skip "no futimes", 6;
     note("check futimes");
     open(my $fh, "<", 'b');
     $foo = (utime $ut,$ut + $delta, $fh);
