diff -up perl-5.38.0-RC2/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/linux.pm.0200~ perl-5.38.0-RC2/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/linux.pm
--- perl-5.38.0-RC2/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/linux.pm.0200~	2023-06-28 19:13:45.736767210 +0200
+++ perl-5.38.0-RC2/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/linux.pm	2023-06-28 19:13:45.736767210 +0200
@@ -0,0 +1,26 @@
+package ExtUtils::CBuilder::Platform::linux;
+
+use strict;
+use ExtUtils::CBuilder::Platform::Unix;
+use File::Spec;
+
+use vars qw($VERSION @ISA);
+$VERSION = '0.280206';
+@ISA = qw(ExtUtils::CBuilder::Platform::Unix);
+
+sub link {
+  my ($self, %args) = @_;
+  my $cf = $self->{config};
+
+  # Link XS modules to libperl.so explicitly because multiple
+  # dlopen(, RTLD_LOCAL) hides libperl symbols from XS module.
+  local $cf->{lddlflags} = $cf->{lddlflags};
+  if ($ENV{PERL_CORE}) {
+    $cf->{lddlflags} .= ' -L' . $self->perl_inc();
+  }
+  $cf->{lddlflags} .= ' -lperl';
+
+  return $self->SUPER::link(%args);
+}
+
+1;
diff -up perl-5.38.0-RC2/MANIFEST.0200~ perl-5.38.0-RC2/MANIFEST
--- perl-5.38.0-RC2/MANIFEST.0200~	2023-06-28 19:13:45.736767210 +0200
+++ perl-5.38.0-RC2/MANIFEST	2023-06-28 19:14:56.752400439 +0200
@@ -3886,6 +3886,7 @@ dist/ExtUtils-CBuilder/lib/ExtUtils/CBui
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/cygwin.pm		CBuilder methods for cygwin
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/darwin.pm		CBuilder methods for darwin
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/dec_osf.pm	CBuilder methods for OSF
+dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/linux.pm		CBuilder methods for Linux
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/os2.pm		CBuilder methods for OS/2
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/Unix.pm		CBuilder methods for Unix
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/VMS.pm		CBuilder methods for VMS
