--- perl-5.16.0/utils/h2ph.PL~	2012-06-22 05:42:23.452592086 +0200
+++ perl-5.16.0/utils/h2ph.PL	2012-06-22 05:42:28.006535152 +0200
@@ -168,10 +168,15 @@ while (defined (my $file = next_file()))
 		    }
 		    print OUT "'sub $name () {$new;}' unless defined(&$name);\n";
 		}
-	    } elsif (/^(include|import|include_next)\s*([<\"])(.*)[>\"]/) {
+	    } elsif (/^(include|import|include_next)\s*([<\"])(.*)([>\"])/) {
+
                 $incl_type = $1;
                 $incl_quote = $2;
                 $incl = $3;
+                if ($4 eq '"') {
+                     $incl = "$1$incl" if $file =~ m!(.*/)! && -e "$1$incl";
+                }
+
                 if (($incl_type eq 'include_next') ||
                     ($opt_e && exists($bad_file{$incl}))) {
                     $incl =~ s/\.h$/.ph/;
