diff -up perl-5.28.2/Makefile.SH.0015~ perl-5.28.2/Makefile.SH
--- perl-5.28.2/Makefile.SH.0015~	2019-04-02 22:36:35.000000000 +0200
+++ perl-5.28.2/Makefile.SH	2019-04-20 22:45:49.174197929 +0200
@@ -74,7 +74,7 @@ true)
 			exeldflags="-Xlinker -headerpad_max_install_names"
 			;;
 		*)
-			shrpldflags="$shrpldflags -install_name \$(shrpdir)/\$@"
+			shrpldflags="$shrpldflags \$(shrpdir)/libperl.${revision}.${patchlevel}.dylib"
 			;;
 		esac
 		;;
@@ -84,13 +84,15 @@ true)
 		;;
 	sunos*)
 		linklibperl="-lperl"
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	netbsd*|freebsd[234]*|openbsd*|dragonfly*|bitrig*)
 		linklibperl="-L. -lperl"
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	interix*)
 		linklibperl="-L. -lperl"
-		shrpldflags="$shrpldflags -Wl,--image-base,0x57000000"
+		shrpldflags="$shrpldflags -Wl,--image-base,0x57000000 -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	aix*)
 		case "$cc" in
@@ -128,6 +130,9 @@ true)
 	    linklibperl='libperl.x'
 	    DPERL_EXTERNAL_GLOB=''
 	    ;;
+	linux*)
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
+	    ;;
 	esac
 	case "$ldlibpthname" in
 	'') ;;
