--- perl-5.16.2/Configure.soname~	2012-12-13 02:52:54.796297642 +0100
+++ perl-5.16.2/Configure	2012-12-13 02:55:21.221135422 +0100
@@ -8210,7 +8210,7 @@ true)
 			# XXX How handle the --version stuff for MAB?
 			;;
 		linux*|gnu*)  # ld won't link with a bare -lperl otherwise.
-			dflt=libperl.$so
+			dflt=libperl.$so.$revision.$patchlevel
 			;;
 		cygwin*) # ld links now against the dll directly
 			majmin="cygperl5_${patchlevel}_${subversion}.${so}"
@@ -8301,7 +8301,7 @@ if "$useshrplib"; then
 		xxx="-Wl,-R$shrpdir"
 		;;
 	bsdos|linux|irix*|dec_osf|gnu*)
-		xxx="-Wl,-rpath,$shrpdir"
+		xxx=""
 		;;
 	next)
 		# next doesn't like the default...
--- perl-5.16.2/installperl.soname~	2012-12-13 02:52:13.625780963 +0100
+++ perl-5.16.2/installperl	2012-12-13 02:52:24.071912058 +0100
@@ -385,7 +385,7 @@ elsif ($Is_Cygwin) { # On Cygwin symlink
     )
 } else {
     # [als] hard-coded 'libperl' name... not good!
-    @corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    @corefiles = <*.h perl*$Config{lib_ext}>;
 
     # AIX needs perl.exp installed as well.
     push(@corefiles,'perl.exp') if $^O eq 'aix';
@@ -407,6 +407,12 @@ foreach my $file (@corefiles) {
 	}
     }
 }
+mkpath("$opts{destdir}$Config{installprefix}/$Config{lib}", $opts{verbose}, 0777);
+copy_if_diff($libperl, "$opts{destdir}$Config{installprefix}/$Config{lib}");
+chmod(0755, "$opts{destdir}$Config{installprefix}/$Config{lib}/$libperl");
+if ($libperl ne "libperl.so") {
+    symlink($libperl, "$opts{destdir}$Config{installprefix}/$Config{lib}/libperl.so");
+}
 
 # Install main perl executables
 # Make links to ordinary names if installbin directory isn't current directory.
--- perl-5.16.2/Makefile.SH.soname~	2012-10-26 04:44:15.000000000 +0200
+++ perl-5.16.2/Makefile.SH	2012-12-13 02:12:45.258022012 +0100
@@ -795,7 +795,8 @@ $(LIBPERL): $& $(obj) $(DYNALOADER) $(LI
 	true)
 		$spitshell >>$Makefile <<'!NO!SUBS!'
 	rm -f $@
-	$(LD) -o $@ $(SHRPLDFLAGS) $(obj) $(DYNALOADER) $(libs)
+	$(LD) -o $@ $(SHRPLDFLAGS) -Wl,--soname=$(LIBPERL) $(obj) $(DYNALOADER) $(libs)
+	[ "$(LIBPERL)" != "libperl.so" ] && ln -s $(LIBPERL) libperl.so
 !NO!SUBS!
 		case "$osname" in
 		aix)
