diff -up perl-5.26.2/t/base/lex.t.omv~ perl-5.26.2/t/base/lex.t
--- perl-5.26.2/t/base/lex.t.omv~	2018-04-16 22:02:40.634236570 +0200
+++ perl-5.26.2/t/base/lex.t	2018-04-16 22:02:53.490237233 +0200
@@ -1,6 +1,6 @@
 #!./perl
 
-print "1..119\n";
+print "1..120\n";
 
 $x = 'x';
 
@@ -158,9 +158,12 @@ my $test = 31;
   ${^TEST}= "splat";
   @{^TEST}= ("foo", "bar");
   %{^TEST}= ("foo" => "FOO", "bar" => "BAR" );
-
+  
   print "not " if "${^TEST}" ne "splat";
   print "ok $test\n"; $test++;
+  
+  print "not " if "${ ^TEST }" ne "splat";
+  print "ok $test\n"; $test++;
 
   print "not " if "${^TEST}[0]" ne "splat[0]";
   print "ok $test\n"; $test++;
