diff -Naur perl-5.20.2/Configure perl-5.20.2.tpg/Configure
--- perl-5.20.2/Configure	2015-03-21 10:35:32.000000000 +0000
+++ perl-5.20.2.tpg/Configure	2015-03-21 10:46:03.613818313 +0000
@@ -8459,7 +8459,7 @@
 			# XXX How handle the --version stuff for MAB?
 			;;
 		*linux*|gnu*)  # ld won't link with a bare -lperl otherwise.
-			dflt=libperl.$so
+			dflt=libperl.$so.$revision.$patchlevel
 			;;
 		cygwin*) # ld links now against the dll directly
 			majmin="cygperl5_${patchlevel}_${subversion}.${so}"
@@ -8550,7 +8550,7 @@
 		xxx="-Wl,-R$shrpdir"
 		;;
 	bsdos|linux|irix*|dec_osf|gnu*|haiku)
-		xxx="-Wl,-rpath,$shrpdir"
+		xxx=""
 		;;
 	next)
 		# next doesn't like the default...
diff -Naur perl-5.20.2/Makefile.SH perl-5.20.2.tpg/Makefile.SH
--- perl-5.20.2/Makefile.SH	2014-12-27 11:49:22.000000000 +0000
+++ perl-5.20.2.tpg/Makefile.SH	2015-03-21 10:48:40.827023988 +0000
@@ -838,7 +838,8 @@
 	true)
 		$spitshell >>$Makefile <<'!NO!SUBS!'
 	rm -f $@
-	$(LD) -o $@ $(SHRPLDFLAGS) $(obj) $(DYNALOADER) $(libs)
+	$(LD) -o $@ $(SHRPLDFLAGS) -Wl,--soname=$(LIBPERL) $(obj) $(DYNALOADER) $(libs)
+	[ "$(LIBPERL)" != "libperl.so" ] && ln -s $(LIBPERL) libperl.so
 !NO!SUBS!
 		case "$osname" in
 		aix)
diff -Naur perl-5.20.2/installperl perl-5.20.2.tpg/installperl
--- perl-5.20.2/installperl	2015-01-10 12:06:37.000000000 +0000
+++ perl-5.20.2.tpg/installperl	2015-03-21 10:47:42.967472538 +0000
@@ -351,7 +351,7 @@
     map { s|^$coredir/||i; } @corefiles = <$coredir/*.*>;
 }
 elsif ($Is_Cygwin) { # On Cygwin symlink it to CORE to make Makefile happy
-    @corefiles = <*.h libperl*.* perl*$Config{lib_ext}>;
+    @corefiles = <*.h perl*$Config{lib_ext}>;
     my $coredll = "$installarchlib/CORE/$libperl";
     my $instcoredll = "$Config{installarchlib}/CORE/$libperl";
     safe_unlink($coredll);
@@ -390,6 +390,13 @@
     }
 }
 
+mkpath("$opts{destdir}$Config{installprefix}/$Config{lib}", $opts{verbose}, 0777);
+copy_if_diff($libperl, "$opts{destdir}$Config{installprefix}/$Config{lib}");
+chmod(0755, "$opts{destdir}$Config{installprefix}/$Config{lib}/$libperl");
+if ($libperl ne "libperl.so") {
+    symlink($libperl, "$opts{destdir}$Config{installprefix}/$Config{lib}/libperl.so");
+}
+
 # Install main perl executables
 # Make links to ordinary names if installbin directory isn't current directory.
 
