--- cpan/Pod-Perldoc/lib/Pod/Perldoc/ToMan.pm.orig	2012-04-25 02:18:31.000000000 +0200
+++ cpan/Pod-Perldoc/lib/Pod/Perldoc/ToMan.pm	2012-05-22 16:00:21.601982560 +0200
@@ -245,7 +245,8 @@
 	# solves many people's problems.  But I also hear that some mans
 	# don't have a -c switch, so that unconditionally adding it here
 	# would presumably be a Bad Thing   -- sburke@cpan.org
-    push @render_switches, '-c' if( $self->_is_roff and $self->is_cygwin );
+    # [Pixel] our man do have the -c switch, using it!!
+    push @render_switches, '-c';
 
 	return @render_switches;
 	}
