--- perl-5.14.2/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm.linkperl~	2012-03-13 06:20:48.579350597 +0100
+++ perl-5.14.2/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm	2012-03-13 06:21:39.365419481 +0100
@@ -953,7 +953,7 @@ $(INST_DYNAMIC): $(OBJECT) $(MYEXTLIB) $
     }
 
     push @m, sprintf <<'MAKE', $ld_run_path_shell, $ldrun, $ldfrom, $libs;
-	%s$(LD) %s $(LDDLFLAGS) %s $(OTHERLDFLAGS) -o $@ $(MYEXTLIB)	\
+	%s$(LD) %s $(LDDLFLAGS) %s $(OTHERLDFLAGS) -L$(PERL_ARCHLIB)/CORE -lperl -pthread -o $@ $(MYEXTLIB)	\
 	  $(PERL_ARCHIVE) %s $(PERL_ARCHIVE_AFTER) $(EXPORT_LIST)	\
 	  $(INST_DYNAMIC_FIX)
 MAKE
