--- perl-5.14.2/Makefile.SH.soname~	2012-03-13 07:49:19.832289061 +0100
+++ perl-5.14.2/Makefile.SH	2012-03-13 07:51:41.774436006 +0100
@@ -762,7 +762,7 @@ $(LIBPERL): $& $(obj) $(DYNALOADER) $(LI
 	true)
 		$spitshell >>$Makefile <<'!NO!SUBS!'
 	rm -f $@
-	$(LD) -o $@ $(SHRPLDFLAGS) $(obj) $(DYNALOADER) $(libs)
+	$(LD) -o $@ $(SHRPLDFLAGS) -Wl,--soname=$(LIBPERL) $(obj) $(DYNALOADER) $(libs)
 !NO!SUBS!
 		case "$osname" in
 		aix)
